test/libsolidity/semanticTests/immutable/complexGetter.sol

```diff
@@ -0,0 +1,12 @@
+contract C {
+    function() external returns (uint, uint) immutable public x = this.f;
+    function f() external pure returns (uint, uint) {
+        return (1, 2);
+    }
+
+    function test() external returns (uint, uint) {
+        return this.x()();
+    }
+}
+// ----
+// test() -> 1, 2
```

Comment:
Maybe add a direct call to `x` as well?

Reply:
It will return the address which can change depending on the test setup.

Reply:
ah right